dev/doc/build-system.dune.md

```diff
@@ -218,9 +219,9 @@ enables stronger compiler optimizations.
 
 ## OPAM file generation
 
-`.opam` files are automatically generated by Dune from the package
+`.opam` files will be automatically generated by Dune from the package
 descriptions in the `dune-project` file; see Dune's manual for more
-details.
+details. For now we have disabled this due to some bugs.
```

Comment:
What is the exact reason we disabled this?

Reply:
Dune bugs, in particular `dune subst` was broken for source tree with unicode files. We shall re-enable soon, hopefully.

Reply:
This has been fixed, lets generate opam again.

Reply:
That requires a dune version bump which may have other implications, I suggest we do that in a different PR, this PR is large enough as it is.

Reply:
I've created an issue for it https://github.com/coq/coq/issues/16071